app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/FriendsFragment.kt

```diff
@@ -70,85 +81,88 @@ class FriendsFragment(private val database: Database) : Fragment(R.layout.fragme
             return
         }
 
-        // TODO: Migrate away from UserModel and refactor
-        // Initialize the ViewModel
         val userAccountFuture = database.getUserData(currentUser.getUid())
 
         userAccountFuture.thenApply { userdata ->
-
-            // Initialize the ViewModel with the userModel
             val factory = FriendsViewModelFactory(userdata.userId!!, this.database)
             viewModel = ViewModelProvider(this, factory)[FriendsViewModel::class.java]
 
-            // Update the adapter with the fetched data
-            recyclerView.adapter = friendsListAdapter
```

Comment:
Just to make sure I understand, we do not update the adapter as it was already assigned on line 59 (65), right ?

Reply:
correct